applications/SubsCrypt.md

```diff
@@ -0,0 +1,185 @@
+They also have to once set their sha256(token+passphrase) for using the SubsCrypt dashboard without a wallet.
+The authentication with a wallet is checked by the blockchain address sender.
+
+**First milestone**: we will design(it is almost done right now) and implement our smart contract, and we will also design XD files of UI, and we will write our white-paper in this milestone(including the test unit).
+
+The main functions that will be deployed on the chain in this milestone are as follow:
+
+
+| Function | Description | Params | Returns | State mutability | 
+| ------------- | ------------- | ------------- | ------------- | ------------- |
+| addPlan | This function is for providers to add their plans; each plan has duration, price, max refund percent that they are willing to lock in contract and withdraw after that the subscription period has finished. | list of durations, list of prices, list of max refund percent | None | change state |
+| editPlan | This function is for providers to edit their plan. (Old subscriptions are not affected by this change) | index of their plan, new duration, new max refund percent, new price | None | change state |
+| changeDisable | This function is for providers to edit their plan that changes the active or deactivate status of their plan(so people can or can't subscribe in that plan) | plan index| None | change state |
+| subscribe | This payable function is for users to subscribe to their desired service and plan; they have to provide a hash of their password (the auth mechanism will be explained thoroughly in Auth Section) and provider address and plan index and some metadata that is encrypted by the public key of the provider(users can trust providers to share their data with but nobody else can know that data) | provider address, plan index, the hash of pass, An optional encrypted metadata| None | change state(payable) |
+| refund | This function is for users to refund their subscribe anytime they want and instantly withdraw the rest of their money(maximum amount of refund is indicated by max refund percent that provider had set for that plan) | provider address, plan index| None |  change state |
+| withdraw | This function is for providers to withdraw the amount that is now ready to withdraw(this is the money that we lock in the contract when a user subscribes to a plan according to max refund percent, and when their plan is finished, that money can be withdrawn). We used an optimized LinkedList solution, which is really cheap to execute and fast. | None | amount of money you are paid  | change state |
```

Comment:
The "optimized LinkedList solution" sounds interesting - what do you mean by this?

Reply:
We wrote an optimized library based on the Linked-List structure to handle this specific use-case. [library source code](https://github.com/oxydev/SubsCrypt-solidity/blob/main/LinkedList.sol) 
Users can refund part of their money. and we have to keep that amount in the contract. when the plan finishes, blocked money is available for provider.
provider calls "withdraw" and with linkedList we calculate the amount that revielded.
`function withdraw() public returns (uint256 paid) {
        require(providers[msg.sender].initiated,'You are not a registered provider');
        paid = providers[msg.sender].paymentManager.process(now.sub(startTime) / 86400  );
        if (paid > 0)
            msg.sender.transfer(paid);
        return paid;
    }`